### PR TITLE
python311Packages.compressai: init at 1.2.4

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Add [compressai](https://github.com/InterDigitalInc/CompressAI), a PyTorch library and evaluation platform for end-to-end compression research.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/compressai/default.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  nativeBuildInputs = [
+    pybind11
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    ipywidgets
+    jupyter
```
Comment (NickCao) [inline pkgs/development/python-modules/compressai/default.nix:42]: These optional dependencies should go to `passthru.optional-dependencies`

File: pkgs/development/python-modules/compressai/default.nix [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    matplotlib
+    pillow
+    pytorch-msssim
+    torch
+    torchvision
```
Comment (NickCao) [inline pkgs/development/python-modules/compressai/default.nix:45]: The dependency list still does not look quite right: https://github.com/InterDigitalInc/CompressAI/blob/v1.2.4/setup.py#L133-L138

File: pkgs/development/python-modules/compressai/default.nix [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    export HOME=$(mktemp -d)
+  '';
+
+  nativeCheckInputs = [
+    plotly
+    pytestCheckHook
+  ];
+
+  disabledTests = [
```
Comment (NickCao) [inline pkgs/development/python-modules/compressai/default.nix:72]: Add a comment about why these tests are disabled.

File: pkgs/development/python-modules/compressai/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/compressai/default.nix]: Fixed

File: pkgs/development/python-modules/compressai/default.nix (hunk 3 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/compressai/default.nix:74]: Done

**Review by NickCao — APPROVED** [review]
Result of `nixpkgs-review pr 263552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.compressai</li>
    <li>python310Packages.compressai.dist</li>
    <li>python310Packages.pytorch-msssim</li>
    <li>python310Packages.pytorch-msssim.dist</li>
    <li>python311Packages.compressai</li>
    <li>python311Packages.compressai.dist</li>
    <li>python311Packages.pytorch-msssim</li>
    <li>python311Packages.pytorch-msssim.dist</li>
  </ul>
</details>

**Pull request merged**